### ISSUE TITLE
Build for raspberry pi

### ISSUE THREAD

**Issue by jarvisteach** [petasis/tkdnd#9]
I was wondering if you are still supporting this project? I'm interested in using tkdnd on the Raspberry Pi, the current linux builds don't work, I'm assuming it's because of the different CPU architecture. Could you offer some help on how to build tkdnd on the Pi?

**Comment by petasis** [comment]
Hi,

I am still supporting this project, but I am not sure how to handle raspberry pi. Is it running some sort of X? Is it using the XDND protocol?

**Comment by jarvisteach** [comment]
Great news!

Yes, raspberry pi's default OS is raspbian - just a modded version of debian, so I assume it's XDND...
I think it just needs to be rebuilt for the ARM processor - it's not X86

I have access to lots of Pis, just need some help building it - I tried the various scripts in the distribution, but don't know much about C or TK so struggled with the error messages.

**Comment by petasis** [comment]
Is it easy to install CMake on this OS? Is gcc available? Also, is there a binary, stubs enabled distribution of Tcl, along with the include files?

**Comment by petasis** [comment]
The process to compile it is:
1) Checkout the git head from the github repository.
2) Install CMake & Tcl.
3) Make sure that when writing "tclsh" on the command line, you get the tclsh from the Tcl distribution you want to use for compiling tkdnd.
4) cd in the cmake directory of tkdnd.
5) If raspberry pi is 32 bit, execute in the cmake directory "bash build.sh". If it is 64 bit, execute "bash build64.sh". This will actually invoke cmake, with some parameters. If everything is ok, you will see a directory like "release*" created.
6) Cd in this directory, and execute make. If no errors are generated, execute "make install".
7) The new package will be installed in tkdnd/cmake/runtime directory. You have to manually copy it in your Tcl installtion.

**Comment by petasis** [comment]
If you get any errors, you can report them here, in this tread. Please note that tkdnd needs also Tk (and its public header files).

**Comment by jarvisteach** [comment]
Right - I think everything is installed, when I run build.sh it all looks good until this:
```
-- ===========================================================
--  Welcome to the tkdnd 2.9 build system!
--   * Selected generator:  Unix Makefiles
--   * Operating System ID: Linux-4.4.50-v7+-armv7l
--   * Installation Directory: /home/pi/Downloads/tkdnd-master/cmake/runtime
-- ===========================================================
-- Searching for Tcl/Tk...
-- Found Tclsh: /usr/bin/tclsh (found version "8.6") 
CMake Error at /usr/share/cmake-3.6/Modules/FindPackageHandleStandardArgs.cmake:148 (message):
  Could NOT find TCL (missing: TCL_INCLUDE_PATH)
Call Stack (most recent call first):
  /usr/share/cmake-3.6/Modules/FindPackageHandleStandardArgs.cmake:388 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.6/Modules/FindTCL.cmake:222 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:128 (FIND_PACKAGE)
```

So, something is missing from TCL...

**Comment by jarvisteach** [comment]
I did a search for libtcl, and found it at: /usr/lib/arm-linux-gnueabihf/libtcl8.5.so

**Comment by petasis** [comment]
Where is tcl.h located? You need both tcl.h & Tk.h to compile tkdnd.

**Comment by jarvisteach** [comment]
Hmm, ok - can't find either of those...

I did an apt-get install tcl, but I guess that don;t install those - how do get them?

**Comment by petasis** [comment]
Since your tclsh is of version 8.6, you need something like libtclstub8.6.a & libtkstub8.6.a. Version 8.5 of these libraries will also work.

**Comment by petasis** [comment]
Look for tcl-dev tk-dev or something...

**Comment by jarvisteach** [comment]
Brilliant - I've installed both of those, and now build.sh works

I've now got a libtkdnd2.9.so file!

**Comment by jarvisteach** [comment]
And it works with the basic.tcl and dndSpy.tcl in the demos directory

**Comment by jarvisteach** [comment]
One final question - why is it all badged as tkdnd2.9 - can can i badge it as 2.8?

**Comment by petasis** [comment]
You can change in line 25, tkdnd/CMakeLists.txt from 9 to 8, and rebuild.

The git head has a new addition for some type under windows, that will be included in the next release (which will be 2.9).

**Comment by jarvisteach** [comment]
Thank you so, so much! I now have drag and drop working in my python apps on raspberry pi!!!
